cmake/bintools/llvm/target.cmake

```diff
@@ -7,12 +7,25 @@ if(DEFINED TOOLCHAIN_HOME)
   set(find_program_binutils_args PATHS ${TOOLCHAIN_HOME})
 endif()
 
-find_program(CMAKE_AR      llvm-ar      ${find_program_clang_args}   )
-find_program(CMAKE_NM      llvm-nm      ${find_program_clang_args}   )
-find_program(CMAKE_OBJDUMP llvm-objdump ${find_program_clang_args}   )
-find_program(CMAKE_RANLIB  llvm-ranlib  ${find_program_clang_args}   )
-find_program(CMAKE_STRIP   llvm-strip   ${find_program_clang_args}   )
-find_program(CMAKE_OBJCOPY objcopy      ${find_program_binutils_args})
+# Extract the clang version.  Debian (maybe other distros?) will
+# append a version to llvm-objdump/objcopy
+execute_process(COMMAND ${CMAKE_C_COMPILER} --version
+                OUTPUT_VARIABLE CLANGVER)
+string(REGEX REPLACE "[^0-9]*([0-9]+).*" "\\1" CLANGVER ${CLANGVER})
```

Comment:
I took a look at the llvm --version output and it's a bit disturbing -- multiple lines, the version appears to be in the last word on the first line, but the rest of the line contents aren't fixed (Debian prefixes with 'Debian clang version')? With that in mind, you might make the regex a bit more robust with something like `".* ([0-9]+)\.[^ ]*"` (anything, a space, the version number, a literal dot, anything other than space)

Reply:
Yeah, this isn't super clean.  My first thought was to do something like `echo | clang -E -dM -o - - | fgrep __clang_major__ | awk '{print $3}'`, but cmake's tool integration story is something out of 1970's CP/M and I couldn't make that work without a bunch of separate commands and some extra files at runtime.

As far as this regex, it seems OK to me.  It pulls out the first decimal number in the output, which works to get the major version for all variants of the compiler I could find (tried ones shipped by Ubuntu 20.04, Arch, and Google's internal debian testing distro).

Reply:
yeah, simple is always nice. At least it will *probably* fail instead of giving the wrong answer?